### PR TITLE
Adding Advanced execution notebook (after Alexandra's CR)

### PR DESCRIPTION
**Pull request by NadavClassiq**
### Some notes

- [ ] Please make sure that you placed the files in an appropriate folder
- [ ] And that the files have indicative names.

- [ ] Please note that Classiq runs automatic code linting, which may minorly alter some files.
  - [ ] If you're familiar with `pre-commit`, you may run `pre-commit install`, and then at each commit, your files will be altered in a similar way

### CODE REVIEW

File: tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb [hunk 1]
```diff
@@ -0,0 +1,834 @@
+{
```
Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>A few things that should be made clearer:</p><p> </p><ul><li>what does it mean "with two varying parameters"?-  this confuses me a bit</li><li>Estimetate - calculates the expectation value and the variance of a specified operator for the quantum state determined by the quantum program,</li><li>What is the "Submit" - if it is stated here, it must be explained briefly. </li></ul><p> </p><p>It is good that you survey what's going to be done in the tutorial, but currently it raises many questions. Please elaborate the different points brifley so they can be understood for people that haven't used ExecutionSession before.</p><p> </p><p>Also, is it relevant only for the SDK? if so, also should be stated</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>without the need to re-synthesize</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>can you remove the write_qmod to the end of this section?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>elaborate in a more down to earth manner here. Something like: even if you close your Python environment, and get back to it later, you can retrieve the execution results. Thus, if you are executing on real QPUs or you have an extensive quantum program to emulate, this is a great feature for you.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>nice addition!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>The expectation value is of a specific operator <strong>for</strong> the quantum program.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>please write here the exact notation: <code>estimate(H,qprog)</code> or something like this. Thus to connect the equation to the actual programming.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>You elaborated regarding the decomposition of the state, but not regarding the decomposition of the Hamiltonian in terms of Paulis. This is the interesting part, where several measurements are needed over non-commutative Paulis.</p><p> </p><p>As a first iteration, we can drop this NOTE and we'll add it later on</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>with a</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>Can you give 1 example for a program without parameters before this one?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>what about the variance? I didn't see it in the above <code>EstimationResult</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (EdenSchirmanClassiq) [inline tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>great example</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/608/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

**Comment by EdenSchirmanClassiq** [comment]
Hi @NadavClassiq , 
That's a great (and super important) tutorial!

I had a few notes along the way. Please amend them and push again,
Thanks!

File: tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb (hunk 1 above)
Comment (NadavClassiq) [reply tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>I hope it clearer now.</p>

Comment (NadavClassiq) [reply tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>hope it is clear now.</p>

Comment (NadavClassiq) [reply tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>I didn't understand the note, so I tried to clarify things.</p>

Comment (NadavClassiq) [reply tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>I also made an effort to improve it in this itreation.</p>

Comment (NadavClassiq) [reply tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>Not sure that it is possible (for some reason....), by the documentation, the second argument is not the qprog but the execution pramaters. </p>

Comment (NadavClassiq) [reply tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>The notation <code>estimate(H,qprog)</code>is not correct.</p>

Comment (NadavClassiq) [reply tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>I don't know why I wrote it, it seems like it doesn't exist. I will check it</p>

Comment (NadavClassiq) [reply tutorials/documentation_materials/classiq_101/classiq_concepts/execute/advanced_execution/advanced_execution_user_guide.ipynb:1]: <p>The second argument is optional, it is required only in the parametric case.</p>

**Review by EdenSchirmanClassiq — APPROVED** [review]
Good.

**Pull request closed without merging**